sdk/Pulumi/Provider/Provider.cs

```diff
@@ -531,34 +633,31 @@ private CancellationTokenSource GetToken(ServerCallContext context)
 
         // Helper to deal with the fact that at the GRPC layer any Struct property might be null. For those we just want to return empty dictionaries at this level.
         // This keeps the PropertyValue.Marshal clean in terms of not handling nulls.
-        private ImmutableDictionary<string, PropertyValue> Marshal(Struct? properties)
+        private ImmutableDictionary<string, PropertyValue> Unmarshal(Struct? properties, IDictionary<string, ISet<Urn>>? inputDependencies = default)
```

Comment:
I've renamed this in https://github.com/pulumi/pulumi-dotnet/pull/334. You'll want to rebase.